### PR TITLE
convert ts button types to overloads

### PR DESCRIPTION
**Pull request by jaketrent**
Better HTML attr checking. Lame "El" aliases.

### CODE REVIEW

File: packages/button/src/react/index.tsx [hunk 1]
```diff
@@ -128,29 +128,33 @@ interface ButtonStatics {
   iconAligns: typeof vars.iconAligns
   sizes: typeof vars.sizes
 }
-
 interface BaseButtonProps {
   appearance?: ValueOf<typeof vars.appearances>
+  // TODO: rm? dup of html attr
   disabled?: boolean
```
Comment (danethurber) [inline packages/button/src/react/index.tsx:134]: i think only button has `disabled` - so you'd want to leave it so the anchors can also be disabled

**Review by danethurber — APPROVED** [review]

File: packages/button/src/react/index.tsx [hunk 2]
```diff
@@ -128,29 +128,33 @@ interface ButtonStatics {
...
+type ButtonProps = ButtonAnchorProps | ButtonButtonProps
+
+type ButtonComponent = RefForwardingComponent<
+  ButtonProps,
+  HTMLAnchorElement | HTMLButtonElement,
+  ButtonStatics
+> & {
+  (props: ButtonAnchorProps, ref?: RefObject<HTMLAnchorElement>): JSX.Element
+  (props: ButtonButtonProps, ref?: RefObject<HTMLButtonElement>): JSX.Element
+}
```
Comment (danethurber) [inline packages/button/src/react/index.tsx:157]: the linter doesn't seem to like this style of overriding

File: packages/button/src/react/index.tsx (hunk 1 above)
Comment (jaketrent) [reply packages/button/src/react/index.tsx:134]: good call

File: packages/button/src/react/index.tsx (hunk 2 above)
Comment (jaketrent) [reply packages/button/src/react/index.tsx:156]: Huh, seems to have made it through the last ci lint ok.  What is flagging for you?

Comment (danethurber) [reply packages/button/src/react/index.tsx:156]: ci was erroring out. if that passed i'm not concerned

Comment (jaketrent) [reply packages/button/src/react/index.tsx:156]: Maybe I checked too fast.  We'll see this time around.  CI looked like a test failure to me, which was in Banner, tsc-related.  Maybe there was something else in there too...

**Pull request merged**